src/name_core.jl

```diff
@@ -162,26 +162,36 @@ This is the type of name combination used for binary array operations.
 Where the dimensions of both arrays must be the same.
 """
 function unify_names(names_a, names_b)
-    # 0-Allocations if inputs are the same
-    # 0-Allocation, if has a `:_` see  `@btime (()->unify_names((:a, :b), (:a, :_)))()`
-
+    # @btime (()->unify_names((:a, :b), (:a, :_)))()
     names_a === names_b && return names_a
+    length(names_a) == length(names_b) || incompatible_dimension_error(names_a, names_b)
+
+    ret = try_unify_names(names_a, names_b)
+    ret === nothing && incompatible_dimension_error(names_a, names_b)
 
-    # Error message should not include names until it is thrown, as othrwise
-    # the interpolation allocates and slows everything down a lot.
-    length(names_a) != length(names_b) && incompatible_dimension_error(names_a, names_b)
+    return compile_time_return_hack(ret)
+end
 
+function try_unify_names(names_a, names_b)
     ret = ntuple(length(names_a)) do ii  # remove :_ wildcards
         a = getfield(names_a, ii)
         b = names_b[ii]
         a === :_ && return b
         b === :_ && return a
         a === b && return a
-
-        return false  # mismatch occured, we mark this with a nonSymbol result
+        return false  # mismatch occured, we mark this with a non-Symbol result
     end
-    ret isa Tuple{Vararg{Symbol}} || incompatible_dimension_error(names_a, names_b)
-    return compile_time_return_hack(ret)
+    if ret isa Tuple{Vararg{Symbol}}
+        return compile_time_return_hack(ret)
+    else
+        return nothing
+    end
+end
+
+function names_are_unifiable(names_a, names_b)
+    names_a === names_b && return true
+    length(names_a) == length(names_b) || return false
```

Comment:
if we moved the above two into `try_unify_names`  then we coujld delete the above two lines